packages/vite/src/node/plugins/worker.ts

```diff
@@ -177,41 +145,34 @@ export async function workerFileToUrl(
   id: string,
   query: Record<string, string> | null
 ): Promise<string> {
-  const workerMap = workerCache.get(config)!
-
-  let hash = workerMap.bundle.get(id)
-  if (hash) {
-    // rewrite truth id, no need to replace by asset plugin
-    return config.base + workerMap.emitted.get(hash)!
+  const workerMap = workerCache.get(config.rawConfig || config)!
```

Comment:
Why is `rawConfig` needed? Wouldn't the worker cache work against `config` here too?

Reply:
https://github.com/poyoho/vite/blob/e06b7c06b2fb413c9575232e72df9c9e29bd511b/packages/vite/src/node/config.ts#L524-L529

nested worker use workerConfig, and the top level worker plugin. and the top level will use workerCache use the `rawConfig`.

https://github.com/poyoho/vite/blob/e06b7c06b2fb413c9575232e72df9c9e29bd511b/packages/vite/src/node/plugins/worker.ts#L251-L262

so we need to use `rawConfig` for cache map key.

Reply:
I think the name `raw` was confusing me here. What about naming it `mainConfig`? And I think you could mark it as `@internal` for now, until we have a good use case for this new field out of Vite's implementation.